core/utils/typecast.py

```diff
@@ -101,16 +102,17 @@ def date(x):
     # checks yyyymmdd format in integers
     if isinstance(x, int):
         if len(str(x)) == 8:
-            parsed_date = yyyy_mm_dd(x)
+            parsed_date = yyyy_mm_dd(x, '%Y%m%d')
     elif isinstance(x, str):
+        x = x.strip()  # because DOB Permits Issued has a trailing space at the end... :(
```

Comment:
most of these are linting (sorry!) and some minor refactors but here's the real fix.

the NYC DOB changed the format of their dates sometime in 2020 from `MM/DD/YYYY` to `YYYY-MM-DD ` -- with a TRAILING SPACE at the end. As a result i suspect a few hundred thousand entries don't have dates on them. fixed now though!